File: Assets/Scripts/Utility/AABBi.cs
using UnityEngine;
using System;

namespace Utility
{
	public struct AABBi
	{
		public Vec2i bl, tr;
		
		public AABBi (Vec2i bl, Vec2i tr)
		{
			this.bl = bl;
			this.tr = tr;
		}

		public void FitWhole(Vec2i point)
		{
			if (point.x < bl.x){
				bl.x = point.x;
			}
			if (point.y < bl.y){
				bl.y = point.y;
			}
			
			if (point.x >= tr.x){
				tr.x = point.x + 1;
			}
			if (point.y >= tr.y){
				tr.y = point.y + 1;
			}
		}
		
		public void Fit(Vec2i point)
		{
			if (point.x < bl.x){
				bl.x = point.x;
			}
			if (point.y < bl.y){
				bl.y = point.y;
			}
			
			if (point.x > tr.x){
				tr.x = point.x;
			}
			if (point.y > tr.y){
				tr.y = point.y;
			}
		}

		public void FitAABB(ref AABBi aabb)
		{
			bl.x = Mathf.Min(aabb.bl.x, bl.x);
			bl.y = Mathf.Min(aabb.bl.y, bl.y);

			tr.x = Mathf.Max(aabb.tr.x, tr.x);
			tr.y = Mathf.Max(aabb.tr.y, tr.y);
		}
	}
}

File: SW_Server/AABBi.cpp
#include "stdafx.h"
#include "AABBi.h"


namespace SW_Server
{
	AABBi::AABBi(glm::ivec2 bl, glm::ivec2 tr)
	{
		AABBi::bl = bl;
		AABBi::tr = tr;
	}


	AABBi::~AABBi()
	{
	}

	void AABBi::FitWhole(glm::ivec2 point)
	{
		bl.x = glm::min(point.x++, bl.x);
		bl.y = glm::min(point.y++, bl.y);
		tr.x = glm::max(point.x, tr.x);
		tr.y = glm::max(point.y, tr.y);
	}

	void AABBi::Fit(glm::ivec2 point)
	{
		bl.x = glm::min(point.x, bl.x);
		bl.y = glm::min(point.y, bl.y);
		tr.x = glm::max(point.x, tr.x);
		tr.y = glm::max(point.y, tr.y);
	}

	void AABBi::FitAABB(AABBi & aabb)
	{
		bl.x = glm::min(aabb.bl.x, bl.x);
		bl.y = glm::min(aabb.bl.y, bl.y);

		tr.x = glm::max(aabb.tr.x, tr.x);
		tr.y = glm::max(aabb.tr.y, tr.y);
	}
}
